index.bs

```diff
@@ -11285,45 +11317,55 @@ being thrown.  When not in strict mode, the assignment attempt will be ignored.
+[=Static operations=] are exposed of the [=interface object=].
+[=Regular operations=] are exposed on the [=interface prototype object=],
+unless the operation is [=unforgeable=] or
+the interface was declared with the [{{Global}}] [=extended attribute=],
+in which case they are exposed on every object that implements the interface.
+
+<div algorithm>
+    To <dfn>define the regular operations</dfn> of [=interface=] or [=namespace=] |definition| on |target|, given [=Realm=] |realm|, 
+    run the following steps:
 
-    Note: that is, even if an [=includes statement=] was used
-    to make an operation available on the interface,
-    we pass in the [=interface=] which includes the [=interface mixin=],
-    and not the [=interface mixin=] on which the operation was originally declared.
+    1.  Let |operations| be the [=list=] of [=regular operations=] that are [=members=] of |definition|.
+    1.  [=list/Remove=] from |operations| all the [=operations=] that are [=unforgeable=].
+    1.  [=Define the operations=] |operations| of |definition| on |target| given |realm|.
+</div>
 
-<p class="advisement">
-    The above description has some bugs, especially around partial interfaces. See
-    <a href="https://github.com/heycam/webidl/issues/164">issue #164</a>.
-</p>
+<div algorithm>
+    To <dfn>define the static operations</dfn> of [=interface=] or [=namespace=] |definition| on |target|, given [=Realm=] |realm|, 
+    run the following steps:
+
+    1.  Let |operations| be the [=list=] of [=static operations=] that are [=members=] of |definition|.
+    1.  [=Define the operations=] |operations| of |definition| on |target| given |realm|.
+</div>
 
-For [=namespaces=], the properties corresponding to each declared operation are described in
-[[#namespace-object]]. (We hope to eventually move interfaces to the same explicit
-property-installation style as namespaces.)
+<div algorithm>
+    To <dfn>define the unforgeable regular operations</dfn> of [=interface=] or [=namespace=] |definition| on |target|, given [=Realm=] |realm|, 
+    run the following steps:
+
+    1.  Let |operations| be the [=list=] of [=unforgeable=] [=regular operations=] that are [=members=] of |definition|.
+    1.  [=Define the operations=] |operations| of |definition| on |target| given |realm|.
+</div>
+
+<div algorithm>
+    To <dfn>define the operations</dfn> |operations| of [=interface=] or [=namespace=] |definition| on |target|,
+    given [=Realm=] |realm|, run the following steps:
+
+    1.  [=list/For each=] [=operation=] |op| of |operations|:
+        1.  If |op| is not [=exposed=] in |realm|, then [=iteration/continue=].
+        1.  Let |method| be the result of [=creating an operation function=]
+            given |op|, |definition|, and |realm|.
+        1.  Let |modifiable| be <emu-val>false</emu-val> if |op| is [=unforgeable=]
+            and <emu-val>true</emu-val> otherwise.
+        1.  Let |id| be |op|'s [=identifier=].
```

Comment:
Tiny nit: the order of this step and the next is switched here vs. for attributes. Might as well make them consistent.

Reply:
Fixed in 5585b02.